### PR TITLE
Add SkyMask.fill_region method

### PR DESCRIPTION
**Pull request by OlgaVorokh**
Add `SkyMask.fill_region` method and test (see #650)

@cdeil review this, please.

### CODE REVIEW

**Comment by cdeil** [comment]
@OlgaVorokh - Thanks!

I discussed with in-person with @adonath  just now.
We're still not sure about the best API and behaviour, especially concerning the question of whether `data` should be modified in-place or whether a new mask should be returned and the original object untouched.
We tried writing down a few use cases and at the end had a slight favour for returning a new object.

You currently have this:
```
mask = region.contains(coords)
self.data[mask == True] = 1
self.data[mask == False] = 0
```
That's the same as this by the way (except for maybe modifying the dtype)
```python
self.data = region.contains(coords)
```

Please put this instead:
```python
data = region.contains(coords)
return SkyMask(data=data, wcs=self.wcs)
```

For the test, please add one that has a pixel coordinate region as input in addition.
You could start with a pixel circle region, transform it to a sky circle region, and then check that the output mask from either is the same.

Concerning the expected result, I find this hard to grok:
```
expected_result[:-1, 1:-1] = 1
```
Can you instead please choose an example where it's easy to compute in your head which pixels you expect with 1 and 0 and then put the full expected_result array in the test?
E.g. the one you have in the examples section in the docstring is already a good test case, no?

**Comment by OlgaVorokh** [comment]
@cdeil , I think, we should better impliment method as one of the `init`-like class method, for example, as `SkyMask.empty`. But in this case , mask will be constructed on the base of uploaded region and image. The intitling `fill_region` means on my opinion the changes of  currient data, that's why my suggestion  would be to change it to `over_region`, for example. With the intitling like that it'll obvious that as a result of creation it will be a new object of `SkyMask` class.I don't know the future usages of this method, but if I will be user,  I'll be waiting the creation of the new `SkyMask` object from this method based on currient image and region.  In the other words, I see the usage of this method in the following manner:

```
image = ... some "SkyImage" obj
region = ... some region
mask = SkyMask.over_region(image, region)
```

File: gammapy/image/tests/test_mask.py [hunk 1]
```diff
@@ -42,14 +42,14 @@ def test_fill_region():
...
-                                [0, 1, 1]])
+    expected_result = [[0, 1, 1],
+                       [0, 1, 1]]
     assert_equal(pix_mask.data, expected_result)
 
     sky_region = pix_region.to_sky(wcs=mask.wcs)
     sky_mask = mask.fill_region(sky_region)
-    expected_result = np.array([[0, 1, 1],
-                                [1, 1, 1]])
+    expected_result = [[0, 1, 1],
```
Comment (OlgaVorokh) [inline gammapy/image/tests/test_mask.py]: @cdeil as you  can see, `sky_mask.data` and `pix_mask.data` are not the same.

**Comment by cdeil** [comment]
Rebase, please.

**Comment by cdeil** [comment]
@OlgaVorokh - Thanks!

I've added one commit https://github.com/gammapy/gammapy/commit/f728810c468508eafba22a205a49ba2523366692 locally and merged this PR into master.


@OlgaVorokh - The commit is a result of your comment above that `fill_region` isn't a good name anymore now that the method doesn't touch `self.data`.
I decided to rename and move `SkyMask.fill_region` to `SkyImage.region_mask`, so that it can be called on any image.

Now it works like this:
```python
image = some image
regions = some region
mask = image.region_mask(region)
```
which is close to your proposal from the last comment, except that I don't have to do the extra `SkyMask` import, i.e. I find this a little more convenient to use.

@OlgaVorokh and @adonath - I hope that API is OK with you!?
If no, let me know which one you prefer and why.

**Comment by adonath** [comment]
I'd still prefer to have the region method on `SkyMask` (just because I feel like it's the most natural place to put it right now, in terms of how the code is organised). E.g with an API like so:
```
mask = SkyMask.empty_like(image)
mask = mask.add_region(region)
```
Where `add_region()` would have the default behaviour of `&=`. 

But the current solution is fine with me as well for now. I think it's more important to start using this functionality and check how the API behaves in practice. I'd might be, that we change it in the future anyway (E.g. when the `SkyMask` class is found to be not a useful abstraction or the `SkyImage` name space just grows to much...)

**Pull request merged**